src/main/java/com/example/service/shoppingcart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,71 @@
+package com.example.service.shoppingcart;
+
+import com.example.dto.cartitem.CartItemRequestDto;
+import com.example.dto.cartitem.CartItemUpdateDto;
+import com.example.dto.shoppingcart.ShoppingCartResponseDto;
+import com.example.mapper.CartItemMapper;
+import com.example.mapper.ShoppingCartMapper;
+import com.example.model.CartItem;
+import com.example.model.ShoppingCart;
+import com.example.model.User;
+import com.example.repository.book.BookRepository;
+import com.example.repository.cartitem.CartItemRepository;
+import com.example.repository.shoppingcart.ShoppingCartRepository;
+import java.util.NoSuchElementException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final BookRepository bookRepository;
+
+    @Override
+    @Transactional
+    public ShoppingCartResponseDto save(User user, CartItemRequestDto requestDto) {
+        CartItem cartItem = cartItemMapper.toModel(requestDto);
+        cartItem.setBook(bookRepository.findById(requestDto.getBookId())
+                .orElseThrow(() -> new NoSuchElementException("Cant find")));
```

Comment:
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Cant find")));
```

provide informative message with parameters
check other places to throw such exception